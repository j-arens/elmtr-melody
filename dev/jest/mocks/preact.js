module.exports = {
    unmountComponent: jest.fn(),
};
