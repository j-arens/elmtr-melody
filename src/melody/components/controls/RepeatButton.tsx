import BaseButton from '@components/BaseButton/';
import Icon from '@components/Icon/';
import { toggleRepeat } from '@redux/actions';
import { Action, State } from '@redux/type';
import { h } from 'preact';
import { connect } from 'preact-redux';
const s = require('./styles.scss');

const mapStateToProps = (state: State) => ({
    repeat: state.repeat,
});

const mapDispatchToProps = dispatch => ({
    toggleRepeat: () => dispatch(toggleRepeat()),
});

interface Props {
    repeat: boolean;
    toggleRepeat: () => Action;
}

const RepeatButton = ({ repeat, toggleRepeat }: Props) => {
    const defaultClass = `${s.playbackCtrl} melody-playbackCtrl`;
    const activeClass = `${defaultClass} ${s['playbackCtrl--active']}`;
    return (
        <BaseButton
            key="repeat"
            className={repeat ? activeClass : defaultClass}
            onClick={toggleRepeat}
        >
            <Icon className={s.playbackCtrl__icon} name="repeat" />
        </BaseButton>
    );
};

export default connect(mapStateToProps, mapDispatchToProps)(RepeatButton);
