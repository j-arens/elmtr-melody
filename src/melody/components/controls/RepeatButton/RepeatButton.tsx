import BaseButton from '@components/BaseButton/';
import Icon from '@components/Icon/';
import { WithOptionalClassName } from '@components/type';
import { Action } from '@redux/type';
import { cySelector } from '@utils/index';
import { h } from 'preact';
const s = require('../styles.scss');

export interface StateProps {
    repeat: boolean;
}

export interface DispatchProps {
    toggleRepeat: () => Action;
}

type Props = StateProps & DispatchProps & WithOptionalClassName;

export default ({ repeat, toggleRepeat, className = '' }: Props) => {
    const defaultClass = `${className} ${s.playbackCtrl} melody-playbackCtrl melody-repeat`;
    const activeClass = `${defaultClass} ${s['playbackCtrl--active']} melody-c-repeat-on`;
    return (
        <BaseButton
            key="repeat"
            className={repeat ? activeClass : defaultClass}
            onClick={toggleRepeat}
            {...cySelector('ctrl')}
        >
            <Icon className={s.playbackCtrl__icon} name="repeat" />
        </BaseButton>
    );
};
