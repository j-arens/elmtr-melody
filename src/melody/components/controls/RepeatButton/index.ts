import { WithOptionalClassName } from '@components/type';
import { toggleRepeat } from '@redux/actions';
import { State } from '@redux/type';
import { connect } from 'preact-redux';
import RepeatButton, { DispatchProps, StateProps } from './RepeatButton';

const mapStateToProps = (state: State) => ({
    repeat: state.repeat,
});

const mapDispatchToProps = dispatch => ({
    toggleRepeat: () => dispatch(toggleRepeat()),
});

export default connect<
    StateProps,
    DispatchProps,
    WithOptionalClassName
>(mapStateToProps, mapDispatchToProps)(RepeatButton);
