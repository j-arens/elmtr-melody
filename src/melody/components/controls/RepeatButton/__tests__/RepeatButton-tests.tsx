import { h } from 'preact';
import { render } from 'preact-render-to-string';
import RepeatButton from '../RepeatButton';

describe('RepeatButton', () => {
    it('renders', () => {
        const component = render(
            <RepeatButton
                repeat={false}
                className="atlantis-blue"
            />,
        );
        expect(component).toMatchSnapshot();
    });

    it('renders with a active class if repeat is on', () => {
        const component = render(
            <RepeatButton
                repeat
                className="atlantis-blue"
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
