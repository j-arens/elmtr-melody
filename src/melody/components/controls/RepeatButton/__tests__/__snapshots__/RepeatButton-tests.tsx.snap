// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepeatButton renders 1`] = `
<button data-cy="ctrl" class="btn atlantis-blue playbackCtrl melody-playbackCtrl melody-repeat"
role="button">
  <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
    <path d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"></path>
  </svg>
</button>
`;

exports[`RepeatButton renders with a active class if repeat is on 1`] = `
<button data-cy="ctrl" class="btn atlantis-blue playbackCtrl melody-playbackCtrl melody-repeat playbackCtrl--active melody-c-repeat-on"
role="button">
  <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
    <path d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"></path>
  </svg>
</button>
`;
